src/App.tsx

```diff
@@ -1,10 +1,64 @@
 /* eslint-disable jsx-a11y/control-has-associated-label */
-import React from 'react';
+import React, { useEffect, useState } from 'react';
+
+import { ErrorNotification } from './components/ErrorNotification';
+import { FooterMenu } from './components/FooterMenu';
+import { ListOfTodos } from './components/ListOfTodos';
+import { Header } from './components/Header';
 import { UserWarning } from './UserWarning';
 
-const USER_ID = 0;
+import { getTodos } from './api/todos';
+
+import { Todo } from './types/Todo';
+import { Filter } from './types/Filter';
+import { CustomError } from './types/CustomError';
+import { ActiveTodoData } from './types/ActiveTodoData';
+
+import { USER_ID } from './utils/fetchClient';
+import { initData } from './constants/initData';
 
 export const App: React.FC = () => {
+  const [todos, setTodos] = useState<Todo[]>(initData.todos);
+  const [filter, setFilter] = useState<Filter>(initData.filter);
+  const [customError, setError]
+    = useState<CustomError | null>(initData.customError);
+  const [activeTodoData, setActiveTodo]
+    = useState<ActiveTodoData>(initData.activeTodoData);
+
+  const filterCallback = ((todo: Todo) => {
+    switch (filter) {
+      case Filter.Completed:
+        return todo.completed;
+      case Filter.Active:
+        return !todo.completed;
+      default:
+        return true;
+    }
+  });
+
+  useEffect(() => {
+    getTodos(USER_ID)
+      .then(todosFromServer => {
+        setTodos(todosFromServer);
+        setActiveTodo({
+          hasActiveTodo: todosFromServer.some(todo => !todo.completed),
```

Comment:
`hasActiveTodo` should be boolean. so it would be better to store it properly
```suggestion
          hasActiveTodo: !!todosFromServer.some(todo => !todo.completed),
```